test/catch_test.cpp

```diff
@@ -0,0 +1,103 @@
+#include <algorithm>
+#define CATCH_CONFIG_MAIN
+#include <catch.hpp>
+
+#include <autocheck/sequence.hpp>
+#include <autocheck/check.hpp>
+
+namespace ac = autocheck;
+
+struct reverse_prop_t {
+  template <typename Container>
+  bool operator() (Container& xs) const {
+    Container ys(xs);
+    std::reverse(xs.begin(), xs.end());
+    std::reverse(xs.begin(), xs.end());
+    return xs == ys;
+  }
+};
+
+template <typename T>
+void insert_sorted(const T& x, std::vector<T>& xs) {
+  xs.push_back(x);
+  if (xs.size() == 1) return;
+  for (std::vector<int>::reverse_iterator
+      b = xs.rbegin(), a = b++, e = xs.rend(); (b != e) && (*a < *b); ++b, ++a)
+  {
+    std::iter_swap(a, b);
+  }
+}
+
+TEST_CASE("Archetype Testing", "[check]") {
+  ac::catch_reporter rep;
+
+  SECTION("bool"){
+    ac::check<bool>([] (bool x) { return true; }, 100,
+        ac::make_arbitrary<bool>(), rep);
+  }
+
+  SECTION("bool with default parameters."){
+    ac::check<bool>([] (bool x) { return true; });
+  }
+
+  reverse_prop_t reverse_prop;
+  SECTION("reverse property over int"){
+
+  ac::check<std::vector<int>>(reverse_prop, 100,
```

Comment:
I believe the lines in this `SECTION` and the next two need to be indented another level.